src/hiveboard_bridge/include/hiveboard_bridge/ThreadWrapper.h

```diff
@@ -0,0 +1,34 @@
+#ifndef CATKIN_ROS_THREADWRAPPER_H
+#define CATKIN_ROS_THREADWRAPPER_H
+
+#include "ReceiveAction.h"
+#include "ros/ros.h"
+#include <common/IProtobufStream.h>
+#include <functional>
+#include <hivemind-host/MessageDTO.h>
+#include <optional>
+#include <thread>
+
+/**
+ * A class that listens on a stream and performs some actions when messages arrive
+ */
+class ThreadWrapper {
+  public:
+    /**
+     * Construct a ThreadWrapper object
+     * @param ReceiveAction The action to be performed in the receive thread
+     * \param sleepTimeMs The sleep time between each cycle of the receive thread
```

Comment:
Use @ instead of \ since that's what we use everywhere, you could specify [in] too.